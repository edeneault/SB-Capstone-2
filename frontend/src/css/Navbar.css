@font-face {
  font-family: tesla;
  src: url(./fonts/TESLA.ttf);
}

@font-face {
  font-family: lovely-coffee;
  src: url(./fonts/LovelyCoffee-jEVeO.ttf);
}

.brand-text {
  font-family: lovely-coffee;
  font-size: 2rem;
  color: white;
}

.social {
  background-color: rgb(0, 0, 0);
}

.search {
  background: rgba(147, 147, 147, 0.283);
}

.navbar-toggler:focus,
.navbar-toggler:active,
.navbar-toggler-icon:focus {
  outline: none;
  box-shadow: none;
}

.fixed {
  position: fixed;
  top: 0;
  z-index: 1001;
}

.bg-custom {
  background: rgb(255, 255, 255);
  background: linear-gradient(to bottom, white 65%, rgba(255, 255, 255, 0));
}
