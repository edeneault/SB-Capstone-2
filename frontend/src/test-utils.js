import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

// create a customRender that wraps the UI in a memory Router

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return <MemoryRouter initialEntries={initialRoutes}>{children}</MemoryRouter>;
};

const customRender = (ui, options) => {
  const initialRoutes =
    options && options.initialRoutes ? options.initialRoutes : ["/"];

  return render(ui, {
    wrapper: (args) =>
      MemoryRouterWithInitialRoutes({
        ...args,
        initialRoutes,
      }),
    ...options,
  });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
