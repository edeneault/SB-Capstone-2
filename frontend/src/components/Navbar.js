import React, { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { LinkContainer } from "react-router-bootstrap";
import { Container, Row, Col, Navbar, Nav, NavDropdown } from "react-bootstrap";
import SearchBox from "./SearchBox";
import { logout } from "../actions/userActions";
import coffee_logo from "../coffee_logo.svg";

import "../css/Navbar.css";

const Header = () => {
  const dispatch = useDispatch();
  const [collapse, setCollapse] = useState(true);

  const userLogin = useSelector((state) => state.userLogin);
  const { userInfo } = userLogin;

  const logoutHandler = () => {
    dispatch(logout());
  };

  const handleToggle = () => {
    collapse ? setCollapse(false) : setCollapse(true);
  };

  return (
    <header>
      <Container fluid className='fade-in fixed'>
        <Row>
          <Col lg={12} className='m-0 p-0'>
            <Navbar
              variant='dark'
              expand='xxl'
              collapseOnSelect
              className='social py-0 fs-4 bg-gradient'
            >
              <LinkContainer to='/'>
                <Navbar.Brand className='d-flex mx-2 p-0 brand-text fs-3 '>
                  <img
                    src={coffee_logo}
                    width='68'
                    height='68'
                    className='d-inline-block align-center'
                    alt='impresso-espresso-logo'
                  />
                  <Navbar.Text className='m-0 pt-2 ms-1 text-capitalize brand-text text-white'>
                    IMPRESSO ESPRESSO
                  </Navbar.Text>
                </Navbar.Brand>
              </LinkContainer>

              <Navbar.Toggle
                onToggle={handleToggle}
                aria-controls='search-navbar-nav'
                className='ms-auto border-0'
              >
                <span className='border-0 text-light'>
                  {collapse ? (
                    <i className='fas fa-chevron-down fa-sm pt-1'></i>
                  ) : (
                    <i className='fas fa-chevron-up fa-sm pt-1'></i>
                  )}
                </span>
                <span className='text-dark fs-6 ms-2'>
                  <i className='fas fa-search text-white pt-1'></i>
                </span>
              </Navbar.Toggle>

              <Navbar.Collapse id='search-navbar-nav'>
                <Nav className='ms-auto pt-1'>
                  <SearchBox name='products' />
                </Nav>
              </Navbar.Collapse>
            </Navbar>
          </Col>
        </Row>
        <Row>
          <Col lg={12} className='m-0 p-0'>
            <Navbar
              variant='light'
              expand='lg'
              collapseOnSelect
              className=' py-1 fs-6 bg-custom'
            >
              <Navbar.Toggle aria-controls='basic-navbar-nav' />

              <Navbar.Collapse id='basic-navbar-nav'>
                <Nav className='m-auto'>
                  <LinkContainer className='m-0 ps-3 pe-5' to='/'>
                    <Nav.Link>Home</Nav.Link>
                  </LinkContainer>
                  <LinkContainer className='m-0 ps-3 pe-5' to='/allproducts'>
                    <Nav.Link>Products</Nav.Link>
                  </LinkContainer>
                  <NavDropdown
                    title='Categories'
                    id='categoryMenu'
                    className='m-0 ps-3 pe-5'
                  >
                    <LinkContainer to='/categories/espresso'>
                      <NavDropdown.Item>Espresso</NavDropdown.Item>
                    </LinkContainer>
                    <LinkContainer to='/categories/barista'>
                      <NavDropdown.Item>Barista</NavDropdown.Item>
                    </LinkContainer>
                    <LinkContainer to='/categories/accessories'>
                      <NavDropdown.Item>Accessories</NavDropdown.Item>
                    </LinkContainer>
                  </NavDropdown>

                  <NavDropdown
                    title='Brands'
                    id='brandMenu'
                    className='m-0 ps-3 pe-5'
                  >
                    <LinkContainer to='/brand/Lavazza'>
                      <NavDropdown.Item>Lavazza</NavDropdown.Item>
                    </LinkContainer>
                    <LinkContainer to='/brand/Bremille'>
                      <NavDropdown.Item>Bremille</NavDropdown.Item>
                    </LinkContainer>
                    <LinkContainer to='/brand/Hario'>
                      <NavDropdown.Item>Hario</NavDropdown.Item>
                    </LinkContainer>
                    <LinkContainer to='/brand/illy'>
                      <NavDropdown.Item>Illy</NavDropdown.Item>
                    </LinkContainer>
                    <LinkContainer to='/brand/LuvMuggs'>
                      <NavDropdown.Item>LuvMuggs</NavDropdown.Item>
                    </LinkContainer>
                    <LinkContainer to='/brand/Damesi'>
                      <NavDropdown.Item>Damesi</NavDropdown.Item>
                    </LinkContainer>
                    <LinkContainer to='/brand/BlueMoon'>
                      <NavDropdown.Item>Blue Moon</NavDropdown.Item>
                    </LinkContainer>
                    <LinkContainer to='/brand/Nesgresso'>
                      <NavDropdown.Item>Nesgresso</NavDropdown.Item>
                    </LinkContainer>
                    <LinkContainer to='/brand/FancyBarista'>
                      <NavDropdown.Item>Fancy Barista</NavDropdown.Item>
                    </LinkContainer>
                    <LinkContainer to='/brand/BrightDay'>
                      <NavDropdown.Item>Bright Day</NavDropdown.Item>
                    </LinkContainer>
                    <LinkContainer to='/brand/ItalianCoffee'>
                      <NavDropdown.Item>Italian Coffee</NavDropdown.Item>
                    </LinkContainer>
                  </NavDropdown>
                  <LinkContainer className='m-0 ps-3 pe-5' to='/cart'>
                    <Nav.Link>
                      <i className='fas fa-shopping-cart px-1'></i>Cart
                    </Nav.Link>
                  </LinkContainer>

                  {userInfo ? (
                    <NavDropdown
                      title={userInfo.name}
                      id='username'
                      className='m-0 ps-3 pe-5'
                    >
                      <LinkContainer to='/profile'>
                        <NavDropdown.Item>Profile</NavDropdown.Item>
                      </LinkContainer>
                      <NavDropdown.Item onClick={logoutHandler}>
                        Logout
                      </NavDropdown.Item>
                    </NavDropdown>
                  ) : (
                    <>
                      <LinkContainer to='/login'>
                        <Nav.Link>
                          <i className='fas fa-sign-in-alt'></i> Sign In
                        </Nav.Link>
                      </LinkContainer>
                      <LinkContainer to='/register'>
                        <Nav.Link>
                          <i className='fas fa-sign-out-alt'></i> Register
                        </Nav.Link>
                      </LinkContainer>
                    </>
                  )}
                  {userInfo && userInfo.isAdmin && (
                    <NavDropdown
                      title='Admin'
                      id='adminMenu'
                      className='m-0 px-3'
                    >
                      <LinkContainer to='/admin/userslist'>
                        <NavDropdown.Item>Users</NavDropdown.Item>
                      </LinkContainer>
                      <LinkContainer to='/admin/productslist'>
                        <NavDropdown.Item>products</NavDropdown.Item>
                      </LinkContainer>
                      <LinkContainer to='/admin/orderslist'>
                        <NavDropdown.Item>orders</NavDropdown.Item>
                      </LinkContainer>
                    </NavDropdown>
                  )}
                </Nav>
              </Navbar.Collapse>
            </Navbar>
          </Col>
        </Row>
      </Container>
    </header>
  );
};

export default Header;
