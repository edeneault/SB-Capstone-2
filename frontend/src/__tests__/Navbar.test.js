import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { render } from "../test-utils";
import { initialState } from "../store-mock";

import Navbar from "../components/Navbar";

const mockStore = configureMockStore([thunk]);

test("renders Navbar component without crashing", () => {
  const store = mockStore(initialState);

  const { getByText } = render(
    <Provider store={store}>
      <Navbar />
    </Provider>,
  );

  expect(getByText("IMPRESSO ESPRESSO")).toBeInTheDocument();
  expect(getByText("Home")).toBeInTheDocument();
  expect(getByText("Brands")).toBeInTheDocument();
});

test("matches snapshot", function () {
  const store = mockStore(initialState);
  const { asFragment } = render(
    <Provider store={store}>
      <Navbar />
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
