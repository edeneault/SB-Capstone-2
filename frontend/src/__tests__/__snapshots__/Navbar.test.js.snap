// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <div
      class="fade-in fixed container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="m-0 p-0 col-lg-12"
        >
          <nav
            class="social py-0 fs-4 bg-gradient navbar navbar-expand-xxl navbar-dark"
          >
            <a
              class="d-flex mx-2 p-0 brand-text fs-3  active navbar-brand"
              href="/"
            >
              <img
                alt="impresso-espresso-logo"
                class="d-inline-block align-center"
                height="68"
                src="coffee_logo.svg"
                width="68"
              />
              <span
                class="m-0 pt-2 ms-1 text-capitalize brand-text text-white navbar-text"
              >
                IMPRESSO ESPRESSO
              </span>
            </a>
            <button
              aria-controls="search-navbar-nav"
              aria-label="Toggle navigation"
              class="ms-auto border-0 navbar-toggler collapsed"
              type="button"
            >
              <span
                class="border-0 text-light"
              >
                <i
                  class="fas fa-chevron-down fa-sm pt-1"
                />
              </span>
              <span
                class="text-dark fs-6 ms-2"
              >
                <i
                  class="fas fa-search text-white pt-1"
                />
              </span>
            </button>
            <div
              class="navbar-collapse collapse"
              id="search-navbar-nav"
            >
              <div
                class="ms-auto pt-1 navbar-nav"
              >
                <div
                  class="container"
                >
                  <div
                    class="d-flex py-2 px-3 row"
                  >
                    <div
                      class="col"
                    >
                      <form
                        class="d-flex ms-auto mt-1"
                      >
                        <input
                          class="rounded-start smaller-input form-control form-control-sm"
                          id="search"
                          name="q"
                          placeholder="search products"
                          type="text"
                        />
                        <button
                          class="btn rounded-end smaller-button btn btn-secondary "
                          type="submit"
                        >
                          <i
                            class="fas fa-search"
                          />
                        </button>
                      </form>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </nav>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="m-0 p-0 col-lg-12"
        >
          <nav
            class=" py-1 fs-6 bg-custom navbar navbar-expand-lg navbar-light"
          >
            <button
              aria-controls="basic-navbar-nav"
              aria-label="Toggle navigation"
              class="navbar-toggler collapsed"
              type="button"
            >
              <span
                class="navbar-toggler-icon"
              />
            </button>
            <div
              class="navbar-collapse collapse"
              id="basic-navbar-nav"
            >
              <div
                class="m-auto navbar-nav"
              >
                <a
                  class="m-0 ps-3 pe-5  active nav-link"
                  data-rb-event-key="/"
                  href="/"
                >
                  Home
                </a>
                <a
                  class="m-0 ps-3 pe-5 nav-link"
                  data-rb-event-key="/allproducts"
                  href="/allproducts"
                >
                  Products
                </a>
                <div
                  class="m-0 ps-3 pe-5 nav-item dropdown"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="dropdown-toggle nav-link"
                    href="#"
                    id="categoryMenu"
                    role="button"
                  >
                    Categories
                  </a>
                </div>
                <div
                  class="m-0 ps-3 pe-5 nav-item dropdown"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="dropdown-toggle nav-link"
                    href="#"
                    id="brandMenu"
                    role="button"
                  >
                    Brands
                  </a>
                </div>
                <a
                  class="m-0 ps-3 pe-5 nav-link"
                  data-rb-event-key="/cart"
                  href="/cart"
                >
                  <i
                    class="fas fa-shopping-cart px-1"
                  />
                  Cart
                </a>
                <div
                  class="m-0 ps-3 pe-5 nav-item dropdown"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="dropdown-toggle nav-link"
                    href="#"
                    id="username"
                    role="button"
                  >
                    AdminName
                  </a>
                </div>
                <div
                  class="m-0 px-3 nav-item dropdown"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="dropdown-toggle nav-link"
                    href="#"
                    id="adminMenu"
                    role="button"
                  >
                    Admin
                  </a>
                </div>
              </div>
            </div>
          </nav>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
